File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day21FractalArtTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day21FractalArtTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day21FractalArt("/twenty_seventeen/Day21-FractalArt-input.txt", 5, 18);

        assertEquals(142, dp.getPart1Answer().intValue());
        assertEquals(1879071, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day21FractalArt("/twenty_seventeen/Day21-FractalArt-example#1.txt", 2, 5);

        assertEquals(12, dp.getPart1Answer().intValue());
        assertEquals(0, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day21FractalArtTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day21FractalArtTest  extends FlatSpec with Matchers {
  "2017 Day 21 - Input" should "calculate answers" in {
    val dp = new Day21FractalArt("/twenty_seventeen/Day21-FractalArt-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day21FractalArt("/twenty_seventeen/Day21-FractalArt-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
